src/VecSim/algorithms/hnsw/hnsw.h

```diff
@@ -28,22 +29,16 @@ using std::pair;
 #define HNSW_INVALID_ID    UINT_MAX
 #define HNSW_INVALID_LEVEL SIZE_MAX
 
-// This type is strongly bounded to `idType` because of the way we get the link list:
-//
-// linklistsizeint *neighbours_list = get_linklist_at_level(element_internal_id, level);
-// unsigned short neighbours_count = getListCount(neighbours_list);
-// auto *neighbours = (idType *)(neighbours_list + 1);
-//
-// TODO: reduce the type to smaller type when possible
-typedef idType linklistsizeint;
+typedef uint16_t linklistsizeint;
```

Comment:
Let's seize this opportunity to be consistent... 
```suggestion
typedef uint16_t linkListSize;
```